### ISSUE TITLE
NuxtJs - Navigator is not defined

### ISSUE THREAD

**Issue by Geminii** [handsontable/vue-handsontable-official#70]
Hi everybody, 

We are using handsontable-pro version through the library `@handsontable-pro/vue`. 
I defined my plugin and load it at ssr false but i got this error : 
`navigator is not defined` in the function `setBrowserMeta()` of node_modules/handsontable/commonjs/helpers/browser.js

My plugin file looks like : 
```
import Vue from 'vue'
import HotTable from '@handsontable-pro/vue';

export default HotTable;
```

Thanks a lot for your help.
Best regards,

**Comment by AMBudnik** [comment]
Hi @Geminii 

do you not import the template?
In my case it's

`import SampleApp from './sampleApp.vue';`

**Comment by Geminii** [comment]
Hi @AMBudnik 

I try to create a component and include it in my page but same error. 

I import handson from : `npm install handsontable-pro @handsontable-pro/vue`. All dependencies are ok (handsontable, handsontable pro, handsontable pro vue).
I create a file plugins to use handson table. 

My app.vue looks like : 
```
<template>
  <div id="hot-preview">
    <HotTable :settings="settings"></HotTable>
  </div>
</template>

<script>
  import HotTable from '~/plugins/handsontable';

  export default {
    data: function() {
      return {
        settings: {
          data: [
            ["", "Ford", "Volvo", "Toyota", "Honda"],
            ["2016", 10, 11, 12, 13],
            ["2017", 20, 11, 14, 13],
            ["2018", 30, 15, 12, 13]
          ],
          colHeaders: true,
          rowHeaders: true,
        }
      };
    },
    components: {
      HotTable
    }
  }
</script>
```

In my `nuxt.config.js`, i have include my plugin file in ssr: false. But not working :( 
But the problem seems to come from handsontable core library into `browser.js`.
How can i fix this ? 

I try to add <no-ssr></no-ssr> but not works too :(
Thanks for your help.

**Comment by AMBudnik** [comment]
Sorry for misunderstanding.

Can you try to change 

```
let HotTable = null;

  if (process.browser) {
   HotTable = require('@handsontable-pro/vue')
  }
```

in the `app.vue`
instead of `import HotTable from '~/plugins/handsontable';` 

and then go with 

```
build: {
    vendor: [
      'handsontable'
    ],
  }
```

in the `nuxt.config.js` file

**Comment by Geminii** [comment]
OK i finally did it among to you :) 

If someone is interresting to include handsontable into NuxtJs : 

`nuxt.config.js` file :
```
build: {
  vendor: ['handsontable'],
},
plugins: [
   '~/plugins/vue-handsontable'
]
```

`vue-handsontable.js` file :
```
let HotTable = null;

if (process.client) {
  HotTable = require('@handsontable-pro/vue')
}

export default HotTable
```

`app.vue` file : 
```
<script>
  import HotTable from '~/plugins/vue-handsontable'

  export default {
    data: function() {
      return {
        settings: {
          data: [
            ["", "Ford", "Volvo", "Toyota", "Honda"],
            ["2016", 10, 11, 12, 13],
            ["2017", 20, 11, 14, 13],
            ["2018", 30, 15, 12, 13]
          ],
          colHeaders: true,
          rowHeaders: true,
        }
      };
    },
    components: {
      HotTable
    }
  }
</script>
```

And job is done :) 
Thanks a lot again @AMBudnik for your quickly help !

Have a good day.
Regards,

**Comment by AMBudnik** [comment]
Great! Thanks for an update :)

**Comment by KWeaver87** [comment]
I've ran across this same issue in my project, and following the advice given so far hasn't solved it yet.
For me, the navigator error only happens when I navigate my browser directly to the HOT page. If I go to another page and route into the HOT page, then it works fine.
Also want to note, adding the vendor prop to build gives a deprecation warning, since NuxtJs 2.0.0 uses Webpack 4.

My nuxt.config.js plugins:
```javascript
plugins: [
  '@/plugins/element-ui',
  {
    src: '@/plugins/hot-table',
    ssr: false,
  },
  '@plugins/feathers-vuex',
],
```

My hot-table.js, following Nuxt-given examples from other plugins:
```javascript
import Vue from 'vue'
import { HotTable } from '@handsontable/vue'

export default () => {
  if (process.client) {
    Vue.component('hot-table', HotTable)
  }
}
```

My simplified wrapper component that uses the hot-table component:
```vue
<template>
  <div class="hotWrapper">
    <hot-table
      ref="hotComponentRef"
      :settings="hotSettings"
    />
  </div>
</template>

<script>
export default {
  props: {
    mapping: {
      type: Object,
      required: true,
    },
  },
  data() {
    return {
      hotSettings: {
        columns: this.mapping.columnDefs,
        data: this.mapping.rowData,
      },
    }
  },
}
</script>
```

I'd definitely appreciate any help on figuring this out.

**Comment by AMBudnik** [comment]
Hey @KWeaver87 

sorry for keeping you waiting.

I just got a reply from our CTO that

>User mentions that HoT works when navigating from other subpage, that's when page is rendered by the browser, not server side. Another reason to believe the problem is with SSR. But have to be confirmed.

but if you could share the whole code it will be much easier to debug it.

**Comment by KWeaver87** [comment]
Thanks for the reply! I just pushed the code out to a new repo: https://github.com/KWeaver87/MappingFrontEnd
Apologies if things are messy or hard to follow, I'm still fairly new at development. This is just a side project for an internal tool at work, so I don't have much direct support from my co-workers.

**Comment by AMBudnik** [comment]
Hey @KWeaver87 

I got an error after running the server

![zrzut ekranu 2019-02-8 o 10 32 52](https://user-images.githubusercontent.com/10757813/52469878-fce82980-2b8c-11e9-8893-bbca93ef3902.png)

can you write me at support@handsontabe.com?

**Comment by venkatperi** [comment]
This is what worked for me:

**plugins/vue-handsontable.ts**
```
// @ts-ignore
const HotTable = !process.client ? null : require('@handsontable/vue').HotTable
export default HotTable
```

**nuxt.config.js**
```
build: {
    vendor: [
      'handsontable',
      "vue-handsontable"
    ],
...

 plugins: [
    { src: '~/plugins/vue-handsontable', ssr: false },
...
  ],
```

**Usage in blah.vue**
```
<template>
    import HotTable from "~/plugins/vue-handsontable"
...
```

**Comment by YUND4** [comment]
Atry it this solution but hi have this error 

ailed to compile with 1 errors                                                                                    friendly-errors 14:20:48

This dependency was not found:                                                                                             friendly-errors 14:20:48
                                                                                                                           friendly-errors 14:20:48
* nuxt_plugin_vuehandsontable_755616e5 in ./.nuxt/index.js                                                                 friendly-errors 14:20:48
                                                                                                                           friendly-errors 14:20:48
To install it, you can run: npm install --save nuxt_plugin_vuehandsontable_755616e5

**Comment by Leskd** [comment]
@venkatperi Hi,  I follow your code.  But when I refresh the page, It's still has error info : navigator is not defined. Hope to get your help! thanks

**Comment by udaandev** [comment]
I am implementing handsontable in nuxt js and facing same error. Please help

**Comment by AMBudnik** [comment]
Hey Guys,

please share the code if you're still facing this issue. I'd build it on my device and check what's wrong.

**Comment by wojciechczerniak** [comment]
Missing  `navigator` is a configuration issue with SSR. Please follow this article: https://medium.com/@codebeast_/why-your-third-party-plugin-dont-work-in-nuxt-and-how-to-fix-it-d1a8caadf422 to configure SSR for Handsontable
